### PR TITLE
Error monitoring of the self-hosted installer

### PR DESCRIPTION
**Pull request by ethanhs**
In this PR we:

- Add the ability to send events to sentry (currently just an event on error/exit of non-zero status)
- Print a nice stack trace when there is an error (I have some tricks to make it even nicer, such as moving things into functions)
- Prompt the user on (and save!) their preference of reporting errors to us or not (saved e.g. for future upgrading etc)

This seems to work if I sprinkle a "false" into one of the files in `install/` and the grouping seems to work correctly too.

Fixes https://github.com/getsentry/self-hosted/issues/740

### CODE REVIEW

File: install/_lib.sh
Comment (BYK) [inline install/_lib.sh:39]: This volume mount is very likely to cause issues on WSL and potentially in macOS as they use VMs and network drives. Just beware :)

File: install/error-handling.sh
Comment (BYK) [inline install/error-handling.sh:5]: Really? :D Why not `self-hosted`?

**Review by BYK — APPROVED** [review]
Nice!

**Review by aminvakil — APPROVED** [review]
This will be great! Thanks!

Some questions though:
- Can non-sentry collaborator (myself :) ) view these logs and if yes, how?
- Do we have a sample log of what will be sent?

I'd also suggest adding some uuid (some hash of IP?) which gets printed and gets sent. Then we can ask users their uuid to view their logs on our instance too. I'm not sure about privacy/policy issues this may brought, but as we're already getting users' IP, I don't think a hash of that would be a problem. This can be pasted in github in public too without a problem.

File: install/_lib.sh
Comment (ethanhs) [reply install/_lib.sh:39]: Well, I was testing on a Mac, so at least we should be good there! :)

Could you explain why would it cause issues with the WSL?

File: install/error-handling.sh
Comment (ethanhs) [reply install/error-handling.sh:5]: It's already at `self-hosted.sentry.net`! But I can always change this if you think I should.

**Comment by ethanhs** [comment]
> Can non-sentry collaborator (myself :) ) view these logs and if yes, how?

The sentry instance is not SaaS, so I'll talk to @chadwhitacre about if this would be possible.

> Do we have a sample log of what will be sent?

We don't, but I could generate one. It is the same as the default log we store locally after installation.

> I'd also suggest adding some uuid (some hash of IP?) which gets printed and gets sent.

This is a good idea. I am worried that someone could just enumerate all hashes of IPs (there aren't that many IPv4s) and so it wouldn't be secret. I'll have to think about a better source for the hash but I think this is a good idea overall.

File: install/_lib.sh
Comment (hubertdeng123) [inline install/_lib.sh:11]: Is there a specific reason to move the logic from `detect-platform.sh` here? Thought it might be potentially easier to read if the logic existed in another file

File: install/error-handling.sh
Comment (hubertdeng123) [inline install/error-handling.sh:16]: tiny: `We may retain your OS username, IP address, and installer log for 30 days.`

File: install/error-handling.sh
Comment (hubertdeng123) [inline install/error-handling.sh:7]: is this because our error monitoring will be more detailed so this is not needed?

File: install/error-handling.sh
Comment (BYK) [reply install/error-handling.sh:5]: Ah, didn't notice that :D I'd still prefer something like `installer` or something but of course, up to you :)

File: install/_lib.sh
Comment (BYK) [reply install/_lib.sh:39]: If it works on Mac, I think it will work on other places. This might have been my scar from a different mapping for my drives.

File: install/error-handling.sh
Comment (chadwhitacre) [reply install/error-handling.sh:5]: I like `installer`, fits if we ever get around to adding run-time monitoring in a separate project.

File: install/error-handling.sh
Comment (chadwhitacre) [reply install/error-handling.sh:16]: Good use case for [GitHub suggestions](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request), @hubertdeng123. ;)

```suggestion
  echo "We may retain your OS username, IP address, and installer log for 30 days."
```

**Comment by chadwhitacre** [comment]
> Can non-sentry collaborator (myself :) ) view these logs and if yes, how?

I am checking with Legal to see what we can offer here. At the very least Sentry has a feature for sharing public links to specific events, I don't have a ton of experience with it but once we start collecting data we can experiment with this to see if it's anonymized enough. Here's a contrived example (I'm not finding mention of this in the docs):

https://self-hosted.getsentry.net/share/issue/81cdf4322c86473e8fd7662aac0f8d51/

File: install/error-handling.sh
Comment (ethanhs) [reply install/error-handling.sh:7]: Yes, the stack trace includes the lineno for each part of the stack up to where the error occured.

File: install/_lib.sh
Comment (ethanhs) [reply install/_lib.sh:39]: I know these mounts can be quite finicky so understandable 😅

File: install/_lib.sh
Comment (ethanhs) [reply install/_lib.sh:11]: It is needed below for the `send_event` function, so it had to be moved here.

File: install/error-handling.sh
Comment (ethanhs) [reply install/error-handling.sh]: Done!

File: install/error-handling.sh
Comment (chadwhitacre) [inline install/error-handling.sh:4]: Can we use `self-hosted`? Makes it easier to distinguish between ours and SaaS.

Screen Shot 2022-09-07 at 2 48 51 PM

**Review by hubertdeng123 — APPROVED** [review]

File: install/_lib.sh
Comment (chadwhitacre) [reply install/_lib.sh:11]: Why do we need to define `send_event` in `_lib.sh` instead of in `error-handling.sh`?

File: install/error-handling.sh
Comment (chadwhitacre) [inline install/error-handling.sh:65]: Will this hash the same if I hit the error from an installation at `/home/sentry` and a second at `/usr/local/sentry`?

File: install/error-handling.sh
Comment (chadwhitacre) [inline install/error-handling.sh:47]: Any need to unset after we're done?

File: install/error-handling.sh
Comment (chadwhitacre) [inline install/error-handling.sh:48]: What's `-v`? Is it portable?

File: install.sh
Comment (chadwhitacre) [inline install.sh:2]: What is capital E again?

Comment (chadwhitacre) [reply install.sh:2]: Ethan says it's for subshells and sourced files or something.

File: install.sh
Comment (chadwhitacre) [inline install.sh:17]: What if there's an error in parse cli or dc-detect?

Comment (chadwhitacre) [reply install.sh:17]: Our error handling depends on Docker specifics, so dc detect should come before.

Parse CLI is pretty thin, fine to leave it where it is.

File: install/error-handling.sh
Comment (chadwhitacre) [reply install/error-handling.sh:48]: Sets a variable, should be fine.

File: install/_lib.sh
Comment (chadwhitacre) [inline install/_lib.sh:48]: Can we send full logfile as an attachment?

File: install/error-handling.sh
Comment (chadwhitacre) [inline install/error-handling.sh:59]:
```suggestion
          printf -v traceback '%s\n' "$traceback${indent//a/ }→ $src:$funcname:$lineno"
```

File: install/error-handling.sh
Comment (chadwhitacre) [reply install/error-handling.sh:47]: xtrace enhances the debugging info we get from bash, it's fine to run dc stop (the last thing we do) with this still on

File: install/_lib.sh
Comment (chadwhitacre) [reply install/_lib.sh:48]: No, not trivially given current sentry-cli. Will have to be a follow-on. :-/

**Review by chadwhitacre — CHANGES_REQUESTED** [review]
A few open items. Looking good so far!

File: install/error-handling.sh
Comment (ethanhs) [reply install/error-handling.sh:4]: It seems that I would have to re-create the sentry instance to change this. I can do that if you want.

File: install/error-handling.sh
Comment (ethanhs) [reply install/error-handling.sh:65]: It will, I confirmed this is just the file name not the full path.

File: install/error-handling.sh
Comment (ethanhs) [reply install/error-handling.sh]: Done, changed on both sides, forgot there was a setting for this 😓

File: install/_lib.sh
Comment (ethanhs) [reply install/_lib.sh]: Moved to `error-handling.sh`

File: install.sh
Comment (ethanhs) [reply install.sh:14]: I misremembered where it was, and there was no simple revert, oops! Now corrected.

File: install/error-handling.sh
Comment (chadwhitacre) [inline install/error-handling.sh:33]: This is a lie-ish. The `.reporterrors` file ends up under `install/`. I would expect to be in the distribution root. No?

Comment (ethanhs) [reply install/error-handling.sh:33]: Yeah, this is a good point, since the installer mostly runs things in the `install` folder. I'll change it so it places it in the distribution root.

File: install/error-handling.sh
Comment (chadwhitacre) [inline install/error-handling.sh:22]: This is insufficient, because we aren't guaranteed to be in the `install` directory.

https://github.com/getsentry/self-hosted/blob/013519a46c696438016a1577615d47f1ec677212/install/_lib.sh#L11-L16

I made a PR at #1686 to propose a significant copy update, and it includes a fix using `$basedir`. What do you think?

**Comment by chadwhitacre** [comment]
Trying out the error reporter (with #1686) ...

```diff
diff --git a/install/check-latest-commit.sh b/install/check-latest-commit.sh
index 7f35660..334b942 100644
--- a/install/check-latest-commit.sh
+++ b/install/check-latest-commit.sh
@@ -1,5 +1,7 @@
 echo "${_group}Checking for latest commit ... "
 
+echo "Oh bother!"; exit 1
+
 # Checks if we are on latest commit from github if it is running from master branch
 if [[ -d "../.git" && "${SKIP_COMMIT_CHECK:-0}" != 1 ]]; then
   if [[ $(git branch | sed -n '/\* /s///p') == "master" ]]; then

```

```
$ ./install.sh 
▶ Parsing command line ...

▶ Detecting Docker platform
Detected Docker platform is linux/arm64

▶ Initializing Docker Compose ...

▶ Setting up error handling ...
Found a .reporterrors file. What does it say? yes

▶ Checking for latest commit ... 
Oh bother!
$
```

Will it [show up](https://self-hosted.getsentry.net/organizations/self-hosted/issues/)? 🧐

**Comment by chadwhitacre** [comment]
```diff
diff --git a/install/check-latest-commit.sh b/install/check-latest-commit.sh
index 7f35660..e9248c3 100644
--- a/install/check-latest-commit.sh
+++ b/install/check-latest-commit.sh
@@ -1,5 +1,7 @@
 echo "${_group}Checking for latest commit ... "
 
+false "Oh bother!"
+
 # Checks if we are on latest commit from github if it is running from master branch
 if [[ -d "../.git" && "${SKIP_COMMIT_CHECK:-0}" != 1 ]]; then
   if [[ $(git branch | sed -n '/\* /s///p') == "master" ]]; then

```

```
$ ./install.sh 
▶ Parsing command line ...

▶ Detecting Docker platform
Detected Docker platform is linux/arm64

▶ Initializing Docker Compose ...

▶ Setting up error handling ...
Found a .reporterrors file. What does it say? yes

▶ Checking for latest commit ... 
Error in check-latest-commit.sh:19.
'[[ "$DID_CLEAN_UP" -eq 1 ]]' exited with status 0
-> ./install.sh:main:19
--> check-latest-commit.sh:source:3

Event dispatched: 6ee22387-6644-4fcd-873b-638d6a08c6f6
Cleaning up...
$
```

**Comment by chadwhitacre** [comment]
Hrm, a couple issues there:

1. `Error in check-latest-commit.sh:19.` does not exist, there is no line 19 in that file (even with the patch). From the traceback it looks like lineno is coming from the `main.sh` frame?
2. `check-latest-commit.sh:source:3` is the correct line, but the message is wrong, I expect to see `false "Oh bother\!"` as the message, not `[[ "$DID_CLEAN_UP" -eq 1 ]]`

Looks like the traceback code needs some work.

**Review by chadwhitacre — CHANGES_REQUESTED** [review]
Seeing unexpected behavior in traceback code. Comments in main thread.

File: install/error-handling.sh
Comment (hubertdeng123) [inline install/error-handling.sh:99]: > 1. `Error in check-latest-commit.sh:19.` does not exist, there is no line 19 in that file (even with the patch). From the traceback it looks like lineno is coming from the `main.sh` frame?

The `${BASH_LINENO[$i]}` is the line number in the source file `${BASH_SOURCE[$i+1]}` as seen [here](https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html)

File: install/error-handling.sh
Comment (hubertdeng123) [inline install/error-handling.sh:91]: > 2. `check-latest-commit.sh:source:3` is the correct line, but the message is wrong, I expect to see `false "Oh bother\!"` as the message, not `[[ "$DID_CLEAN_UP" -eq 1 ]]`

cmd here was being overwritten by line 92 beforehand
```
if [[ "$DID_CLEAN_UP" -eq 1 ]]; then
```

**Comment by hubertdeng123** [comment]
This is the result I get now:
```
./install.sh
▶ Parsing command line ...

▶ Detecting Docker platform
Detected Docker platform is linux/arm64

▶ Initializing Docker Compose ...

▶ Setting up error handling ...
Found a .reporterrors file. What does it say? yes

▶ Checking for latest commit ... 
Error in check-latest-commit.sh:3.
'false "Oh bother!"' exited with status 0
-> ./install.sh:main:19
--> check-latest-commit.sh:source:3

Event dispatched: ccb89503-3b67-4f65-b492-c1b56c2c7832
Cleaning up...
```

**Comment by chadwhitacre** [comment]
Why is it saying the exit status is 0, though? That indicates success. 🤔

```
$ false
$ echo $?
1
$
```

File: install/error-handling.sh
Comment (chadwhitacre) [reply install/error-handling.sh:91]: Good find! 👍

**Comment by chadwhitacre** [comment]
Fixed in fee3d5105c168c995321512eeec3216082fcdcd4. ;)

[Here it is in Sentry](https://self-hosted.getsentry.net/organizations/self-hosted/issues/3/?project=3):

Screen Shot 2022-09-09 at 10 54 09 AM

Screen Shot 2022-09-09 at 10 53 43 AM

**Comment by chadwhitacre** [comment]
Not sure the IP is going to be that valuable if it tends to be from an internal block.

**Comment by hubertdeng123** [comment]
> Fixed in [fee3d51](https://github.com/getsentry/self-hosted/commit/fee3d5105c168c995321512eeec3216082fcdcd4). ;)
> 
> [Here it is in Sentry](https://self-hosted.getsentry.net/organizations/self-hosted/issues/3/?project=3):
> 

ah nice catch

File: install/error-handling.sh
Comment (chadwhitacre) [inline install/error-handling.sh:20]: This is confusing to me. Why do we pass `SENTRY_DSN` in the dockerized case but not `SENTRY_ORG` or `SENTRY_PROJECT`?

This also seems like not a great experience on non-linux/amd64, to hard exit during error handling vs. warning early during setup-error-handling and silently passing here. Of course better would be to make it work, how hard would it be to build sentry-cli for ARM? Are we close? Link?

Comment (chadwhitacre) [reply install/error-handling.sh:20]: I'm also not sure of the advantage of hashing the traceback ourselves vs. sending the traceback as the fingerprint and letting Sentry hash.

Do we want separate status codes to hash together as they do now? It seems a little weird to have variable messages in events that are grouped together.

variation

Comment (hubertdeng123) [reply install/error-handling.sh:20]: I think we should have sentry-cli for ARM

https://docs.sentry.io/product/cli/installation/#docker-image

Comment (hubertdeng123) [reply install/error-handling.sh:20]: a bit confused, is there a reason why we're using the dockerized case for `linux/amd64`?

Comment (hubertdeng123) [reply install/error-handling.sh:20]: It's unclear to me why we're not passing the rest of the env variables too.

Comment (hubertdeng123) [reply install/error-handling.sh:20]: actually we only have a pre built image for amd64 😢 

https://hub.docker.com/search?q=sentry-cli

Comment (hubertdeng123) [reply install/error-handling.sh:20]: I'll answer my own question, dockerized case is preferred or else sentry-cli will need to be installed manually and what you mentioned above chad

File: install/error-handling.sh
Comment (hubertdeng123) [inline install/error-handling.sh:9]: notes here after running on ubuntu linux/amd64

This line errors out
```
docker: Error response from daemon: create $(pwd): "$(pwd)" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.
```

File: install/error-handling.sh
Comment (hubertdeng123) [inline install/error-handling.sh:9]: Needed to change `$(pwd)` to `$basedir` here to find the logfile

File: install/error-handling.sh
Comment (hubertdeng123) [reply install/error-handling.sh]: Resolved now that \$(pwd) -> $basedir

**Comment by hubertdeng123** [comment]
This is the sentry event sent from my ubuntu setup:

https://self-hosted.getsentry.net/organizations/self-hosted/issues/3/?query=is%3Aunresolved

**Comment by hubertdeng123** [comment]
Behavior on `amd` and `arm` platform is tested and looks good to me now.

We can put something on the backlog to try and dockerize non `amd` platforms as a future item.

**Comment by chadwhitacre** [comment]
I've updated the non-amd64 case in de504d8c9fb696c2df7a33cf0c454a3477488a0f. Here are the cases we need to check when an error is present (`false 'Oh bother!'` as above):

| platform | sentry-cli | .reporterrors | early warning (n/y) | prompt (n/y) | late warning (n/y) | attempt (n/y/f = failed) |
|:---|:---|:---|:---|:---|:---|:---|
| amd | no | ∅ |  |  |  |  |
| amd | no | no |  |  |  |  |
| amd | no | yes |  |  |  |  |
| amd | yes | ∅ |  |  |  |  |
| amd | yes | no |  |  |  |  |
| amd | yes | yes |  |  |  |  |
| arm | no | ∅ |  |  |  |  |
| arm | no | no |  |  |  |  |
| arm | no | yes |  |  |  |  |
| arm | yes | ∅ |  |  |  |  |
| arm | yes | no |  |  |  |  |
| arm | yes | yes |  |  |  |  |

My plan for testing this (short of writing a test script) would be to fiddle with the `check_for_sentry_cli` and `check_for_linux_amd64` to simulate the alternate cases. Like this:

```diff
diff --git a/install/error-handling.sh b/install/error-handling.sh
index 13902cf..17c1c0f 100644
--- a/install/error-handling.sh
+++ b/install/error-handling.sh
@@ -7,10 +7,10 @@ export SENTRY_PROJECT=installer
 function check_for_linux_amd64 {
   # This is the only platform for which we have a build of sentry-cli.
   # I guess this could theoretically be dynamic (try to fetch a build).
-  test $DOCKER_PLATFORM == "linux/amd64"
+  test $DOCKER_PLATFORM == "linux/arm64"
 }
 function check_for_sentry_cli {
-  command -v sentry-cli &> /dev/null
+  command -v entry-cli &> /dev/null
 }
 function check_for_reportability {
   # I'm sure this isn't the most terse syntax, but it's the one I could get to work.
```

**Comment by chadwhitacre** [comment]
I hate to say it but I think we should check for whether there is a `sentry-cli` Docker image dynamically rather than hard-coding it. That way we can ship this and then build a Docker image for arm64 at a later date decoupled from a code change and it will "just work."

**Comment by hubertdeng123** [comment]
lol, just tried running the `docker run --rm -v $basedir:/work -e SENTRY_ORG=$SENTRY_ORG -e SENTRY_PROJECT=$SENTRY_PROJECT -e SENTRY_DSN=$SENTRY_DSN getsentry/sentry-cli` on my mac M1 and it is working...

Perhaps it was failing in the first place due to syntax issues and we never got around to verifying that. I don't even think we need to include separate logic for arm64

**Comment by hubertdeng123** [comment]
well, pushed up changes! could use some assistance just to confirm that this works

**Comment by chadwhitacre** [comment]
Here's what I get: 

```
$ ./install.sh 
▶ Parsing command line ...

▶ Detecting Docker platform
Detected Docker platform is linux/arm64

▶ Initializing Docker Compose ...

▶ Setting up error handling ...
Found a .reporterrors file. What does it say? yes

▶ Checking for latest commit ... 
Error in check-latest-commit.sh:3.
'false 'Oh bother!'' exited with status 1
-> ./install.sh:main:19
--> check-latest-commit.sh:source:3

Unable to find image 'getsentry/sentry-cli:latest' locally
latest: Pulling from getsentry/sentry-cli
c7ed990a2339: Pulling fs layer
18d3cc651798: Pulling fs layer
3e6a010ae3c1: Pulling fs layer
4e81a4699a2d: Pulling fs layer
0c93296acec2: Pulling fs layer
4e81a4699a2d: Waiting
0c93296acec2: Waiting
18d3cc651798: Verifying Checksum
3e6a010ae3c1: Verifying Checksum
3e6a010ae3c1: Download complete
4e81a4699a2d: Verifying Checksum
4e81a4699a2d: Download complete
c7ed990a2339: Pull complete
18d3cc651798: Pull complete
3e6a010ae3c1: Pull complete
4e81a4699a2d: Pull complete
0c93296acec2: Verifying Checksum
0c93296acec2: Download complete
0c93296acec2: Pull complete
Digest: sha256:9c214c9b56a9292c2b1dc727e096bab185aa23bafaddb290a52377e7f5022157
Status: Downloaded newer image for getsentry/sentry-cli:latest
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Event dispatched: 13f97186-4ffa-4203-8ef6-ce5abdb7128d
Cleaning up...
$
```

What do you think @ethanhs? Is this okay to ship?

File: install/error-handling.sh
Comment (ethanhs) [reply install/error-handling.sh]: Yeah, we should pass SENTRY_ORG and SENTRY_PROJECT to docker as well, this was an oversight (added them after writing this I think 😢 )

Agreed testing up front for sentry-cli missing is better. I haven't gotten much information about sentry-cli's build system but that they are working on it, I can ask if there is a link to follow along on the changes.

**Comment by ethanhs** [comment]
> What do you think @ethanhs? Is this okay to ship?

I think with one or two minor changes!

> Do we want separate status codes to hash together as they do now?

I think this is still an issue? But I'm not sure why it is happening tbh. I wanted to hash the traceback as a means of making the issues grouped based on traceback, but it seems that isn't happening...

**Comment by chadwhitacre** [comment]
I do see [an event showing up](https://self-hosted.getsentry.net/organizations/self-hosted/issues/3/events/13f971864ffa42038ef6ce5abdb7128d/?project=3), so it appears to work despite the warning.

**Comment by chadwhitacre** [comment]
> I think this is still an issue?

I'm not going to block on this. Generally the exit code will likely be stable, it was a bit of a fluke that it varied for us.

> But I'm not sure why it is happening tbh.

I think it's happening because the exit code is not included in the traceback, it only appears in `$cmd_exit`.

**Comment by hubertdeng123** [comment]
To add more context for the warning 
```
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
```

I get that for other images we use in `self-hosted` since we use other `linux/amd64` images, so I think this is fine

**Comment by chadwhitacre** [comment]
We set `DOCKER_PLATFORM` though ...

https://github.com/getsentry/self-hosted/blob/0e1239c78f87f2b810576f6fc7327b6e8c908a85/install/detect-platform.sh#L18-L23

So why does it say, "and no specific platform was requested"?

**Comment by chadwhitacre** [comment]
lolsob `DOCKER_PLATFORM` is unique to us, it's not a Docker envvar. 🙄

**Comment by hubertdeng123** [comment]
What's also interesting is that I don't get that warning when pulling sentry-cli:latest

```
Unable to find image 'getsentry/sentry-cli:latest' locally
latest: Pulling from getsentry/sentry-cli
c7ed990a2339: Pulling fs layer
18d3cc651798: Pulling fs layer
3e6a010ae3c1: Pulling fs layer
4e81a4699a2d: Pulling fs layer
0c93296acec2: Pulling fs layer
4e81a4699a2d: Waiting
0c93296acec2: Waiting
18d3cc651798: Download complete
3e6a010ae3c1: Verifying Checksum
3e6a010ae3c1: Download complete
c7ed990a2339: Verifying Checksum
c7ed990a2339: Download complete
c7ed990a2339: Pull complete
18d3cc651798: Pull complete
3e6a010ae3c1: Pull complete
4e81a4699a2d: Verifying Checksum
4e81a4699a2d: Download complete
4e81a4699a2d: Pull complete
0c93296acec2: Verifying Checksum
0c93296acec2: Download complete
0c93296acec2: Pull complete
Digest: sha256:9c214c9b56a9292c2b1dc727e096bab185aa23bafaddb290a52377e7f5022157
Status: Downloaded newer image for getsentry/sentry-cli:latest
Event dispatched: 3cf243d5-19be-44c8-8c21-c486a5c1d6f3
Cleaning up...
```

**Comment by ethanhs** [comment]
Ok then I think this is ready, @chadwhitacre want to give it the green check-mark?

**Comment by chadwhitacre** [comment]
Oh wow @hubertdeng123 I wonder if the delta between your env and mine is related to what we're seeing on #1671?

I think we're good here though. We use `DOCKER_PLATFORM` to set `platform` in `docker-compose.yml`, but that does not influence the `sentry-cli` image.

**Comment by ethanhs** [comment]
> Oh wow @hubertdeng123 I wonder if the delta between your env and mine is related to what we're seeing on https://github.com/getsentry/self-hosted/pull/1671?

This is what I was thinking as well.

**Comment by chadwhitacre** [comment]
Okay, pushed one more commit 5df2ab645f186044e4af3b809b6ee7b334be990d to fail gracefully, if the `sentry-cli` image doesn't work for some reason let's find out early so we don't muddy the waters with a double failure during error handling.

I'm ready to merge if yinz are!

**Review by chadwhitacre — APPROVED** [review]

**Pull request merged**
